allennlp/data/dataset_readers/huggingface_datasets_reader.py

```diff
@@ -0,0 +1,190 @@
+from typing import Iterable, Optional
+
+from allennlp.data import DatasetReader, Token, Field
+from allennlp.data.fields import TextField, LabelField, ListField
+from allennlp.data.instance import Instance
+from datasets import load_dataset, Dataset
+from datasets.features import ClassLabel, Sequence, Translation, TranslationVariableLanguages
+from datasets.features import Value
+
+
+class HuggingfaceDatasetSplitReader(DatasetReader):
+    """
+    This reader implementation wraps the huggingface datasets package
+    to utilize it's dataset management functionality and load the information in AllenNLP friendly formats
+    Note: Reader works w.r.t to only one split of the dataset,
+    i.e. you would need to create separate reader for separate splits
+
+    Following dataset and configurations have been verified and work with this reader
+
+            Dataset                       Dataset Configuration
+            `xnli`                        `ar`
+            `xnli`                        `en`
+            `xnli`                        `de`
+            `xnli`                        `all_languages`
+            `glue`                        `cola`
+            `glue`                        `mrpc`
+            `glue`                        `sst2`
+            `glue`                        `qqp`
+            `glue`                        `mnli`
+            `glue`                        `mnli_matched`
+            `universal_dependencies`      `en_lines`
+            `universal_dependencies`      `ko_kaist`
+            `universal_dependencies`      `af_afribooms`
+            `afrikaans_ner_corpus`        `NA`
+            `swahili`                     `NA`
+            `conll2003`                   `NA`
+            `dbpedia_14`                  `NA`
+            `trec`                        `NA`
+            `emotion`                     `NA`
+    """
+
+    def __init__(
+        self,
+        max_instances: Optional[int] = None,
+        manual_distributed_sharding: bool = False,
+        manual_multiprocess_sharding: bool = False,
+        serialization_dir: Optional[str] = None,
+        dataset_name: str = None,
+        split: str = "train",
+        config_name: Optional[str] = None,
+    ) -> None:
+        super().__init__(
+            max_instances,
+            manual_distributed_sharding,
+            manual_multiprocess_sharding,
```

Comment:
If you end up utilizing `self.shard_iterable()` (see my other comment) or similar, you can set these to `True`

```suggestion
            manual_distributed_sharding=True,
            manual_multiprocess_sharding=True,
```